### PR TITLE
feat: enable data grid in Chatbot

### LINKED ISSUE #1343: [FEATURE] Integrate data grid into Chatbot by using PPL query.
**Is your feature request related to a problem?**
image
We are developing dashboards assistant, a Chatbot that can show data grid when LLM gives a PPL query.

**What solution would you like?**
Integrate data grid into Chatbot by using the customized render mechanism provided by Chatbot.

**What alternatives have you considered?**
N/A

**Do you have any additional context?**
N/A

### PR DESCRIPTION
**Pull request by SuZhou-Joe**
### Description
image

### Issues Resolved
#1343 

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/event_analytics/explorer/events_views/data_grid.tsx
Comment (joshuali925) [inline public/components/event_analytics/explorer/events_views/data_grid.tsx:45]: nit. oui uses `Oui` in camelcases
```suggestion
  OuiDataGridProps?: Partial<EuiDataGridProps>;
```

File: public/dependencies/register_assistant.tsx
Comment (joshuali925) [inline public/dependencies/register_assistant.tsx:18]: we don't need ppl based visualizations in chat anymore?

**Review by joshuali925 — APPROVED** [review]

File: public/dependencies/register_assistant.tsx
Comment (SuZhou-Joe) [reply public/dependencies/register_assistant.tsx:18]: No, only data grid is needed.

**Comment by SuZhou-Joe** [comment]
@joshuali925 Could we have one more approval to get this PR merged? The registerContentRenderer method has been changed to registerMessageRenderer and we need this PR to get merged.

**Review by YANG-DB — APPROVED** [review]

**Comment by SuZhou-Joe** [comment]
@YANG-DB @joshuali925 Could you please help to merge this PR?

**Comment by SuZhou-Joe** [comment]
@joshuali925 @joshuali925 , could you please add `backport 2.x` label after backport of #1304 get merged?

**Comment by joshuali925** [comment]
sure #1374

**Comment by SuZhou-Joe** [comment]
@joshuali925 @YANG-DB , as backport failed, I create backport PR manually here. https://github.com/opensearch-project/dashboards-observability/pull/1383. Please help to review.

BTW, a minor change(1 line change) in parser, could you please help to review?
https://github.com/opensearch-project/dashboards-observability/pull/1382

**Pull request merged**
